v2/tools/generator/internal/codegen/pipeline/export_controller_type_registrations.go

```diff
@@ -117,67 +190,108 @@ func catalogSecretPropertyChains(def astmodel.TypeDefinition, definitions astmod
 }
 
 type indexFunctionBuilder struct {
-	propChains [][]*astmodel.PropertyDefinition
+	propChains []*propertyChain
 }
 
-type indexFunctionBuilderContext struct {
-	props []*astmodel.PropertyDefinition
+// propertyChain represents an immutable chain of properties that can be used to index a secret on a resource. Each
```

Comment:
`immutable` here threw me because you changed `requiredForPropertyPath` and I kept trying to figure out how this is immutable given that.

I _think_ you mean that the chain part of it is immutable... but wondering if it would be clearer if you just dropped the word immutable here altogether. Though it would be nice if there was a solution where this whole thing is immutable... not sure what that'd look like though.

Reply:
Rephrased, and took out the word _immutable_.